### PR TITLE
Log steps of the panic in new block proposing

### LINKED ISSUE #138: Fix panic in new block proposing
If all of the nodes are connected and blocks are being mined after some time there is a panic in handleEvents goroutine, here is a small log trace:
```
DEBUG[06-20|17:51:03.426|consensus/tendermint/core/core.go:193]         broadcasting                             step=StepProposeDone    msg="{Code: 2, Address: 0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f}"
DEBUG[06-20|17:51:03.426|consensus/tendermint/backend/backend.go:294]   worker loop. resending                   msg="msg hash 0x93977750b35aa47024d7e7958c02f7a1a2bb6dc584517ecccbca17cf6822afb1   length 196 time 2019-06-20 17:51:03.426805282 +0000 UTC m=+181.780187167 peers 4: 0x09428E8674496e2D1E965402F33A9520c5fCBbE2 0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D 0xc443C6c6AE98F5110702921138D840e77dA67702 "
DEBUG[06-20|17:51:03.527|consensus/tendermint/backend/backend.go:238]   inner sender loop                        try=0 peer=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 msg=0x93977750b35aa47024d7e7958c02f7a1a2bb6dc584517ecccbca17cf6822afb1
DEBUG[06-20|17:51:03.527|consensus/tendermint/backend/backend.go:238]   inner sender loop                        try=0 peer=0xc443C6c6AE98F5110702921138D840e77dA67702 msg=0x93977750b35aa47024d7e7958c02f7a1a2bb6dc584517ecccbca17cf6822afb1
DEBUG[06-20|17:51:03.527|consensus/tendermint/backend/backend.go:248]   inner sender loop. success               try=1 peer=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 msg=0x93977750b35aa47024d7e7958c02f7a1a2bb6dc584517ecccbca17cf6822afb1
DEBUG[06-20|17:51:03.527|consensus/tendermint/backend/backend.go:238]   inner sender loop                        try=0 peer=0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D msg=0x93977750b35aa47024d7e7958c02f7a1a2bb6dc584517ecccbca17cf6822afb1
DEBUG[06-20|17:51:03.527|consensus/tendermint/backend/backend.go:248]   inner sender loop. success               try=1 peer=0xc443C6c6AE98F5110702921138D840e77dA67702 msg=0x93977750b35aa47024d7e7958c02f7a1a2bb6dc584517ecccbca17cf6822afb1
DEBUG[06-20|17:51:03.527|consensus/tendermint/backend/backend.go:238]   inner sender loop                        try=0 peer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 msg=0x93977750b35aa47024d7e7958c02f7a1a2bb6dc584517ecccbca17cf6822afb1
DEBUG[06-20|17:51:03.527|consensus/tendermint/backend/backend.go:248]   inner sender loop. success               try=1 peer=0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D msg=0x93977750b35aa47024d7e7958c02f7a1a2bb6dc584517ecccbca17cf6822afb1
DEBUG[06-20|17:51:03.527|consensus/tendermint/backend/backend.go:248]   inner sender loop. success               try=1 peer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 msg=0x93977750b35aa47024d7e7958c02f7a1a2bb6dc584517ecccbca17cf6822afb1
DEBUG[06-20|17:51:03.527|consensus/tendermint/backend/backend.go:294]   worker loop. resending                   msg="msg hash 0x60421432bf6f36ad294bce49d3833d0f7f4b52a65544d7ac52b06b4f06dd5aa7   length 130 time 2019-06-20 17:51:03.527686403 +0000 UTC m=+181.881068274 peers 4: 0x09428E8674496e2D1E965402F33A9520c5fCBbE2 0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D 0xc443C6c6AE98F5110702921138D840e77dA67702 "
DEBUG[06-20|17:51:03.628|consensus/tendermint/backend/backend.go:238]   inner sender loop                        try=0 peer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 msg=0x60421432bf6f36ad294bce49d3833d0f7f4b52a65544d7ac52b06b4f06dd5aa7
DEBUG[06-20|17:51:03.628|consensus/tendermint/backend/backend.go:238]   inner sender loop                        try=0 peer=0xc443C6c6AE98F5110702921138D840e77dA67702 msg=0x60421432bf6f36ad294bce49d3833d0f7f4b52a65544d7ac52b06b4f06dd5aa7
DEBUG[06-20|17:51:03.628|consensus/tendermint/backend/backend.go:248]   inner sender loop. success               try=1 peer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 msg=0x60421432bf6f36ad294bce49d3833d0f7f4b52a65544d7ac52b06b4f06dd5aa7
DEBUG[06-20|17:51:03.628|consensus/tendermint/backend/backend.go:238]   inner sender loop                        try=0 peer=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 msg=0x60421432bf6f36ad294bce49d3833d0f7f4b52a65544d7ac52b06b4f06dd5aa7
DEBUG[06-20|17:51:03.628|consensus/tendermint/backend/backend.go:248]   inner sender loop. success               try=1 peer=0xc443C6c6AE98F5110702921138D840e77dA67702 msg=0x60421432bf6f36ad294bce49d3833d0f7f4b52a65544d7ac52b06b4f06dd5aa7
DEBUG[06-20|17:51:03.628|consensus/tendermint/backend/backend.go:248]   inner sender loop. success               try=1 peer=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 msg=0x60421432bf6f36ad294bce49d3833d0f7f4b52a65544d7ac52b06b4f06dd5aa7
DEBUG[06-20|17:51:03.629|consensus/tendermint/core/prevote.go:127]      MessageEvent(Prevote): Received          from=0xc443C6c6AE98F5110702921138D840e77dA67702 to=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f currentHeight=166 msgHeight=166 currentRound=0 msgRound=0 currentStep=StepPrevoteDone    isProposer=false currentProposer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 isNilMsg=false hash=7aec8c…3fd77a type=Prevote             totalVotes=5 totalNilVotes=0 quorumReject=false totalNonNilVotes=5 quorumAccept=true
DEBUG[06-20|17:51:03.629|consensus/tendermint/backend/backend.go:294]   worker loop. resending                   msg="msg hash 0x62fcf8698fb4383bd01d75c178cafa2841c8fc994fd2a2a9cae6946880833806   length 130 time 2019-06-20 17:51:03.629402144 +0000 UTC m=+181.982784009 peers 4: 0x09428E8674496e2D1E965402F33A9520c5fCBbE2 0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D 0xc443C6c6AE98F5110702921138D840e77dA67702 "
DEBUG[06-20|17:51:03.630|consensus/tendermint/core/precommit.go:105]    MessageEvent(Precommit): Received        from=0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D to=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f currentHeight=166 msgHeight=166 currentRound=0 msgRound=0 currentStep=StepPrevoteDone    isProposer=false currentProposer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 isNilMsg=false hash=7aec8c…3fd77a type=Precommit           totalVotes=1 totalNilVotes=0 quorumReject=false totalNonNilVotes=1 quorumAccept=false
DEBUG[06-20|17:51:03.630|consensus/tendermint/backend/backend.go:294]   worker loop. resending                   msg="msg hash 0xfe963f88a3e31786412400a1dd1f963497df374911e821d6977a84fdc62e4a3f   length 196 time 2019-06-20 17:51:03.630188074 +0000 UTC m=+181.983569962 peers 4: 0x09428E8674496e2D1E965402F33A9520c5fCBbE2 0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D 0xc443C6c6AE98F5110702921138D840e77dA67702 "
DEBUG[06-20|17:51:03.730|consensus/tendermint/backend/backend.go:238]   inner sender loop                        try=0 peer=0xc443C6c6AE98F5110702921138D840e77dA67702 msg=0xfe963f88a3e31786412400a1dd1f963497df374911e821d6977a84fdc62e4a3f
DEBUG[06-20|17:51:03.730|consensus/tendermint/backend/backend.go:248]   inner sender loop. success               try=1 peer=0xc443C6c6AE98F5110702921138D840e77dA67702 msg=0xfe963f88a3e31786412400a1dd1f963497df374911e821d6977a84fdc62e4a3f
DEBUG[06-20|17:51:03.730|consensus/tendermint/backend/backend.go:238]   inner sender loop                        try=0 peer=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 msg=0xfe963f88a3e31786412400a1dd1f963497df374911e821d6977a84fdc62e4a3f
DEBUG[06-20|17:51:03.730|consensus/tendermint/backend/backend.go:248]   inner sender loop. success               try=1 peer=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 msg=0xfe963f88a3e31786412400a1dd1f963497df374911e821d6977a84fdc62e4a3f
DEBUG[06-20|17:51:03.730|consensus/tendermint/backend/backend.go:238]   inner sender loop                        try=0 peer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 msg=0xfe963f88a3e31786412400a1dd1f963497df374911e821d6977a84fdc62e4a3f
DEBUG[06-20|17:51:03.730|consensus/tendermint/backend/backend.go:248]   inner sender loop. success               try=1 peer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 msg=0xfe963f88a3e31786412400a1dd1f963497df374911e821d6977a84fdc62e4a3f
DEBUG[06-20|17:51:03.731|consensus/tendermint/core/precommit.go:105]    MessageEvent(Precommit): Received        from=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f to=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f currentHeight=166 msgHeight=166 currentRound=0 msgRound=0 currentStep=StepPrevoteDone    isProposer=false currentProposer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 isNilMsg=false hash=7aec8c…3fd77a type=Precommit           totalVotes=2 totalNilVotes=0 quorumReject=false totalNonNilVotes=2 quorumAccept=false
DEBUG[06-20|17:51:03.731|consensus/tendermint/backend/backend.go:294]   worker loop. resending                   msg="msg hash 0x93977750b35aa47024d7e7958c02f7a1a2bb6dc584517ecccbca17cf6822afb1   length 196 time 2019-06-20 17:51:03.731399395 +0000 UTC m=+182.084781264 peers 4: 0x09428E8674496e2D1E965402F33A9520c5fCBbE2 0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D 0xc443C6c6AE98F5110702921138D840e77dA67702 "
DEBUG[06-20|17:51:03.731|consensus/tendermint/core/precommit.go:105]    MessageEvent(Precommit): Received        from=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 to=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f currentHeight=166 msgHeight=166 currentRound=0 msgRound=0 currentStep=StepPrevoteDone    isProposer=false currentProposer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 isNilMsg=false hash=7aec8c…3fd77a type=Precommit           totalVotes=3 totalNilVotes=0 quorumReject=false totalNonNilVotes=3 quorumAccept=false
DEBUG[06-20|17:51:03.731|consensus/tendermint/backend/backend.go:294]   worker loop. resending                   msg="msg hash 0xb628f059b0cc06a72cca131f42fb322a701d3b6ac0e58a168afdeb4dece0077e   length 196 time 2019-06-20 17:51:03.731893125 +0000 UTC m=+182.085274974 peers 4: 0x09428E8674496e2D1E965402F33A9520c5fCBbE2 0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D 0xc443C6c6AE98F5110702921138D840e77dA67702 "
DEBUG[06-20|17:51:03.734|eth/fetcher/fetcher.go:631]                    Queued propagated block                  peer=02619de2661819c5                           number=166 hash=7aec8c…3fd77a queued=1
DEBUG[06-20|17:51:03.734|eth/fetcher/fetcher.go:642]                    Importing propagated block               peer=02619de2661819c5                           number=166 hash=7aec8c…3fd77a
DEBUG[06-20|17:51:03.737|core/types/nodes.go:39]                        performing                               node=enode://e766ac390e2d99b559aef773c3656fa8d50df2310496ac26ca6c3fc84e21dabb8a0162cc8e34f938d45e0a8ed04955f8ddf1c380182f8ef17a3f08885064505f@172.25.0.13:30303
DEBUG[06-20|17:51:03.737|core/types/nodes.go:39]                        performing                               node=enode://d73b857969c86415c0c000371bcebd9ed3cca6c376032b3f65e58e9e2b79276fbc6f59eb1e22fcd6356ab95f42a666f70afd4985933bd8f3e05beb1a2bf8fdde@172.25.0.11:30303
DEBUG[06-20|17:51:03.737|core/types/nodes.go:39]                        performing                               node=enode://438a5c2cd8fdc2ecbc508bf7362e41c0f0c3754ba1d3267127a3756324caf45e6546b02140e2144b205aeb372c96c5df9641485f721dc7c5b27eb9e35f5d887b@172.25.0.14:30303
DEBUG[06-20|17:51:03.737|core/types/nodes.go:39]                        performing                               node=enode://1f207dfb3bcbbd338fbc991ec13e40d204b58fe7275cea48cfeb53c2c24e1071e1b4ef2959325fe48a5893de8ff37c73a24a412f367e505e5dec832813da546a@172.25.0.12:30303
DEBUG[06-20|17:51:03.737|core/types/nodes.go:39]                        performing                               node=enode://3ce6c053cb563bfd94f4e0e248510a07ccee1bc836c9784da1816dba4b10564e7be1ba42e0bd8d73c8f6274f8e9878dc13814adb381c823264265c06048b4b59@172.25.0.15:30303
DEBUG[06-20|17:51:03.738|core/state/statedb.go:665]                     Trie cache stats after commit            misses=0 unloads=0
DEBUG[06-20|17:51:03.738|trie/database.go:508]                          Dereferenced trie from memory database   nodes=0 size=0.00B time=680ns   gcnodes=0 gcsize=0.00B gctime=25.1µs   livenodes=61 livesize=10.93kB
DEBUG[06-20|17:51:03.738|core/blockchain.go:1238]                       Inserted new block                       number=166 hash=7aec8c…3fd77a uncles=0 txs=0 gas=0 elapsed=1.283ms   root=4fda9d…dd325f
INFO [06-20|17:51:03.738|core/blockchain_insert.go:74]                  Imported new chain segment               blocks=1 txs=0 mgas=0.000 elapsed=2.577ms   mgasps=0.000 number=166 hash=7aec8c…3fd77a cache=14.77kB
INFO [06-20|17:51:03.738|miner/unconfirmed.go:107]                      🔗 block reached canonical chain          number=159 hash=9c22b7…792f73
INFO [06-20|17:51:03.738|miner/worker.go:990]                           Commit new mining work                   number=167 sealhash=4d9745…f64808 uncles=0 txs=0 gas=0 fees=0 elapsed=319.74µs
INFO [06-20|17:51:03.738|miner/worker.go:540]                           New block Seal request                   hash=4d9745…f64808
DEBUG[06-20|17:51:03.738|core/tx_pool.go:418]                           Reinjecting stale transactions           count=0
DEBUG[06-20|17:51:03.832|consensus/tendermint/backend/backend.go:238]   inner sender loop                        try=0 peer=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 msg=0xb628f059b0cc06a72cca131f42fb322a701d3b6ac0e58a168afdeb4dece0077e
DEBUG[06-20|17:51:03.832|consensus/tendermint/backend/backend.go:238]   inner sender loop                        try=0 peer=0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D msg=0xb628f059b0cc06a72cca131f42fb322a701d3b6ac0e58a168afdeb4dece0077e
DEBUG[06-20|17:51:03.832|consensus/tendermint/backend/backend.go:248]   inner sender loop. success               try=1 peer=0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D msg=0xb628f059b0cc06a72cca131f42fb322a701d3b6ac0e58a168afdeb4dece0077e
DEBUG[06-20|17:51:03.832|consensus/tendermint/backend/backend.go:238]   inner sender loop                        try=0 peer=0xc443C6c6AE98F5110702921138D840e77dA67702 msg=0xb628f059b0cc06a72cca131f42fb322a701d3b6ac0e58a168afdeb4dece0077e
DEBUG[06-20|17:51:03.832|consensus/tendermint/backend/backend.go:248]   inner sender loop. success               try=1 peer=0xc443C6c6AE98F5110702921138D840e77dA67702 msg=0xb628f059b0cc06a72cca131f42fb322a701d3b6ac0e58a168afdeb4dece0077e
DEBUG[06-20|17:51:03.832|consensus/tendermint/backend/backend.go:248]   inner sender loop. success               try=1 peer=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 msg=0xb628f059b0cc06a72cca131f42fb322a701d3b6ac0e58a168afdeb4dece0077e
DEBUG[06-20|17:51:03.832|consensus/tendermint/core/precommit.go:105]    MessageEvent(Precommit): Received        from=0xc443C6c6AE98F5110702921138D840e77dA67702 to=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f currentHeight=166 msgHeight=166 currentRound=0 msgRound=0 currentStep=StepPrevoteDone    isProposer=false currentProposer=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 isNilMsg=false hash=7aec8c…3fd77a type=Precommit           totalVotes=4 totalNilVotes=0 quorumReject=false totalNonNilVotes=4 quorumAccept=true
goroutine 200 [running]:
runtime/debug.Stack(0x419ce6, 0xc0194037c0, 0x419c77)
	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/clearmatics/autonity/consensus/tendermint/core.(*core).handleEvents.func1(0xc00028a6c0)
	/autonity/build/_workspace/src/github.com/clearmatics/autonity/consensus/tendermint/core/handler.go:64 +0x9b
panic(0xeb7340, 0x199fcc0)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/clearmatics/autonity/core/types.(*Block).Hash(0x0, 0x0, 0x0, 0x0, 0x0)
	/autonity/build/_workspace/src/github.com/clearmatics/autonity/core/types/block.go:368 +0x4a
github.com/clearmatics/autonity/consensus/tendermint/core.(*core).commit(0xc00028a6c0)
	/autonity/build/_workspace/src/github.com/clearmatics/autonity/consensus/tendermint/core/core.go:213 +0x87
github.com/clearmatics/autonity/consensus/tendermint/core.(*core).handlePrecommit(0xc00028a6c0, 0xc01bc6b1f0, 0x4, 0x4)
	/autonity/build/_workspace/src/github.com/clearmatics/autonity/consensus/tendermint/core/precommit.go:76 +0x640
github.com/clearmatics/autonity/consensus/tendermint/core.(*core).handleCheckedMsg(0xc00028a6c0, 0xc01bc6b1f0, 0x125a2e0, 0xc007fe7180, 0x4, 0x125a2e0)
	/autonity/build/_workspace/src/github.com/clearmatics/autonity/consensus/tendermint/core/handler.go:200 +0x2bf
github.com/clearmatics/autonity/consensus/tendermint/core.(*core).handleMsg(0xc00028a6c0, 0xc019b31a00, 0xc4, 0xc4, 0x1, 0xc4)
	/autonity/build/_workspace/src/github.com/clearmatics/autonity/consensus/tendermint/core/handler.go:150 +0x320
github.com/clearmatics/autonity/consensus/tendermint/core.(*core).handleEvents(0xc00028a6c0)
	/autonity/build/_workspace/src/github.com/clearmatics/autonity/consensus/tendermint/core/handler.go:87 +0x239
created by github.com/clearmatics/autonity/consensus/tendermint/core.(*core).Start
	/autonity/build/_workspace/src/github.com/clearmatics/autonity/consensus/tendermint/core/handler.go:17 +0x4d
CRIT [06-20|17:51:03.832|consensus/tendermint/core/handler.go:65]       panic in core.handleEvents               panic="runtime error: invalid memory address or nil pointer dereference"

```

### PR DESCRIPTION
**Pull request by JekaMas**
closes https://github.com/clearmatics/autonity/issues/140

### CODE REVIEW

File: consensus/tendermint/core/handler.go [hunk 1]
```diff
@@ -62,11 +62,9 @@ func (c *core) handleEvents() {
 	defer func() {
 		if r := recover(); r != nil {
 			debug.PrintStack()
+
 			c.logger.Crit("panic in core.handleEvents", "panic", r)
 		}
-
-		c.currentRoundState = nil
-		<-c.handlerStopCh
```
Comment (aneequesafdar) [inline consensus/tendermint/core/handler.go]: What issue was this causing?

Comment (JekaMas) [reply consensus/tendermint/core/handler.go]: We wont reach this part of the code in case of panic.

**Comment by JekaMas** [comment]
@aneequesafdar I think I got an interesting thing in the logs

```
DEBUG[06-21|12:12:03.481|consensus/tendermint/core/precommit.go:105]    MessageEvent(Precommit): Received        from=0x09428E8674496e2D1E965402F33A9520c5fCBbE2 to=0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D currentHeight=259 msgHeight=259 currentRound=0 msgRound=0 currentStep=StepPrevoteDone    isProposer=false currentProposer=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f isNilMsg=false hash=b4a159…bd6efc type=Precommit           totalVotes=3 totalNilVotes=0 quorumReject=false totalNonNilVotes=3 quorumAccept=false
DEBUG[06-21|12:12:03.481|consensus/tendermint/backend/backend.go:306]   worker loop. resending                   msg="msg hash 0x78bf89228f67d130450f555726d25fb7778a398dcc2b7991ce93fe9733aee362   length 197 time 2019-06-21 12:12:03.481865386 +0000 UTC m=+257.512837673 peers 4: 0x09428E8674496e2D1E965402F33A9520c5fCBbE2 0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f 0xc443C6c6AE98F5110702921138D840e77dA67702 "
DEBUG[06-21|12:12:03.485|eth/fetcher/fetcher.go:631]                    Queued propagated block                  peer=85776bc3e28c4de8                           number=259 hash=b4a159…bd6efc queued=1
DEBUG[06-21|12:12:03.485|eth/fetcher/fetcher.go:642]                    Importing propagated block               peer=85776bc3e28c4de8                           number=259 hash=b4a159…bd6efc
DEBUG[06-21|12:12:03.490|core/types/nodes.go:39]                        performing                               node=enode://e766ac390e2d99b559aef773c3656fa8d50df2310496ac26ca6c3fc84e21dabb8a0162cc8e34f938d45e0a8ed04955f8ddf1c380182f8ef17a3f08885064505f@172.25.0.13:30303
DEBUG[06-21|12:12:03.490|core/types/nodes.go:39]                        performing                               node=enode://3ce6c053cb563bfd94f4e0e248510a07ccee1bc836c9784da1816dba4b10564e7be1ba42e0bd8d73c8f6274f8e9878dc13814adb381c823264265c06048b4b59@172.25.0.15:30303
DEBUG[06-21|12:12:03.490|core/types/nodes.go:39]                        performing                               node=enode://1f207dfb3bcbbd338fbc991ec13e40d204b58fe7275cea48cfeb53c2c24e1071e1b4ef2959325fe48a5893de8ff37c73a24a412f367e505e5dec832813da546a@172.25.0.12:30303
DEBUG[06-21|12:12:03.490|core/types/nodes.go:39]                        performing                               node=enode://438a5c2cd8fdc2ecbc508bf7362e41c0f0c3754ba1d3267127a3756324caf45e6546b02140e2144b205aeb372c96c5df9641485f721dc7c5b27eb9e35f5d887b@172.25.0.14:30303
DEBUG[06-21|12:12:03.490|core/types/nodes.go:39]                        performing                               node=enode://d73b857969c86415c0c000371bcebd9ed3cca6c376032b3f65e58e9e2b79276fbc6f59eb1e22fcd6356ab95f42a666f70afd4985933bd8f3e05beb1a2bf8fdde@172.25.0.11:30303
DEBUG[06-21|12:12:03.490|core/state/statedb.go:665]                     Trie cache stats after commit            misses=0 unloads=0
DEBUG[06-21|12:12:03.490|trie/database.go:508]                          Dereferenced trie from memory database   nodes=0 size=0.00B time=1.771µs gcnodes=0 gcsize=0.00B gctime=131.415µs livenodes=61 livesize=10.93kB
DEBUG[06-21|12:12:03.491|core/blockchain.go:1238]                       Inserted new block                       number=259 hash=b4a159…bd6efc uncles=0 txs=0 gas=0 elapsed=1.892ms   root=4fda9d…dd325f
INFO [06-21|12:12:03.491|core/blockchain_insert.go:74]                  Imported new chain segment               blocks=1 txs=0 mgas=0.000 elapsed=3.664ms   mgasps=0.000 number=259 hash=b4a159…bd6efc cache=14.77kB
DEBUG[06-21|12:12:03.491|core/tx_pool.go:418]                           Reinjecting stale transactions           count=0
INFO [06-21|12:12:03.491|miner/worker.go:990]                           Commit new mining work                   number=260 sealhash=2be9e7…acfcec uncles=0 txs=0 gas=0 fees=0 elapsed=526.304µs
INFO [06-21|12:12:03.491|miner/worker.go:540]                           New block Seal request                   hash=2be9e7…acfcec
DEBUG[06-21|12:12:03.532|consensus/tendermint/backend/backend.go:241]   inner sender loop                        try=0 peer=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f msg=0x78bf89228f67d130450f555726d25fb7778a398dcc2b7991ce93fe9733aee362
DEBUG[06-21|12:12:03.532|consensus/tendermint/backend/backend.go:241]   inner sender loop                        try=0 peer=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 msg=0x78bf89228f67d130450f555726d25fb7778a398dcc2b7991ce93fe9733aee362
DEBUG[06-21|12:12:03.532|consensus/tendermint/backend/backend.go:251]   inner sender loop. success               try=1 peer=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f msg=0x78bf89228f67d130450f555726d25fb7778a398dcc2b7991ce93fe9733aee362
DEBUG[06-21|12:12:03.532|consensus/tendermint/backend/backend.go:241]   inner sender loop                        try=0 peer=0xc443C6c6AE98F5110702921138D840e77dA67702 msg=0x78bf89228f67d130450f555726d25fb7778a398dcc2b7991ce93fe9733aee362
DEBUG[06-21|12:12:03.532|consensus/tendermint/backend/backend.go:251]   inner sender loop. success               try=1 peer=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 msg=0x78bf89228f67d130450f555726d25fb7778a398dcc2b7991ce93fe9733aee362
DEBUG[06-21|12:12:03.532|consensus/tendermint/backend/backend.go:251]   inner sender loop. success               try=1 peer=0xc443C6c6AE98F5110702921138D840e77dA67702 msg=0x78bf89228f67d130450f555726d25fb7778a398dcc2b7991ce93fe9733aee362
DEBUG[06-21|12:12:03.532|consensus/tendermint/core/precommit.go:105]    MessageEvent(Precommit): Received        from=0x4AD219b58a5b46A1D9662BeAa6a70DB9F570deA5 to=0x4B07239Bd581d21AEfcdEe0c6dB38070F9A5FD2D currentHeight=259 msgHeight=259 currentRound=0 msgRound=0 currentStep=StepPrevoteDone    isProposer=false currentProposer=0x850C1Eb8D190e05845ad7F84ac95a318C8AaB07f isNilMsg=false hash=b4a159…bd6efc type=Precommit           totalVotes=4 totalNilVotes=0 quorumReject=false totalNonNilVotes=4 quorumAccept=true
DEBUG[06-21|12:12:03.532|consensus/tendermint/core/core.go:289]         Starting new Round                       Height=260 Round=0
ERROR[06-21|12:12:03.532|consensus/tendermint/core/core.go:215]         commit a NIL block                       block=<nil>                                                                                                                                                                                                                                                                     height=260 round=0
```

1. The validator was on 259 height, collecting preCommits
2. Than `Inserted new block 259` without a quorum
3. Than `Commit new mining work number=260`
4. Received preCommit message for `259`, got a quorum
5. Start new height `Starting new Round Height=260 Round=0`
6. `commit a NIL block `block=<nil> height=260 round=0`
7. panic

I'm going to rename this PR into `log empty proposed blocks` and create a proper issue for it.
And I think that block mining doesn't work properly.

**Comment by aneequesafdar** [comment]
This is very interesting, maybe the conditions in calling c.commit are not implemented properly, or do you think this is something outside of the tendermint package (I mean when you refer to the mining doesn't work properly)?

**Comment by JekaMas** [comment]
@aneequesafdar I think it's all about timeouts, step changing and
```
	latestPendingUnminedBlock   *types.Block
	latestPendingUnminedBlockMu *sync.RWMutex
	unminedBlockCh              chan struct{}
```

**Comment by aneequesafdar** [comment]
ok, I will look into it as well

**Comment by aneequesafdar** [comment]
Did you intend to push `coverage.out` or was it an accident?

**Pull request merged**